iphone/Maps/Core/TrackRecorder/TrackRecordingManager.swift

```diff
@@ -0,0 +1,154 @@
+enum TrackRecordingState: Equatable {
+  case inactive
+  case active
+  case error(TrackRecordingError)
+}
+
+enum TrackRecordingAction {
+  case start
+  case stop
+}
+
+enum TrackRecordingError: Error {
+  case locationIsProhibited
+}
+
+typealias TrackRecordingStateHandler = (TrackRecordingState) -> Void
+
+protocol TrackRecordingObservation {
```

Comment:
Is it needed?

Reply:
At present time not. Ill remove it. I will return this when needed